src/platform/window.cpp

```diff
@@ -59,13 +64,17 @@ Window::Window(ContextType type) {
 
 	_window = glfwCreateWindow(1920, 1080, "", nullptr, nullptr);
 
+	glfwSetCursorPos(_window, 960, 540);
```

Comment:
Why do you set the starting position of the cursor to the mid of the screen? You could easily determine the position (or if it is out of the window by testing this directly after the window creation.

Reply:
IIRC, than line was written before I've implemented mouse enter/leave callbacks and it is indeed unnecessary now. I'll remove it once I start working on the feedback.